fix(client): Error handling of `fetchHttpStream`

teogeb opened the pull request:
If the HTTP `fetch` stream fails, it now propagates the error to the piped stream (the stream which transforms chunks of texts to lines). The underlying stream throws an error e.g. if we call `AbortController#abort()`. 

Before this PR the error in the underlying stream would just stop the output of that stream, but the transform stream didn't stop as it was not notified by the error. The `AsyncGenerator` never completed in that case.

Also replaced `parseError` functionality: the `fetchHttpStream` now throws an explicit `FetchHttpStreamResponseError` if the response is not ok (i.e. HTTP 200) instead of calling `parseError` to form the error.

teogeb committed:
test server util

teogeb committed:
fetchHttpStream returns lines

teogeb committed:
test case

teogeb committed:
error in fetch stream throws an error in AsyncGenerator

teogeb committed:
test cases

teogeb committed:
stream is always defined

teogeb committed:
improve readability

harbu commented:
> changelog?
- Fix resend freeze / stuck issue (https://github.com/streamr-dev/network/pull/1554)

harbu reviewed (approved):
Related to your open question, I think it is unconventional for the `fetchHttpStream` to take in a function for transforming errors in the first place. Instead I would just catch the error in the place where it is invoked and then do the transformation there if any error is thrown.

teogeb committed:
changelog

teogeb committed:
replace parseError functionality from fetchHttpStream

teogeb commented:
> Related to your open question, I think it is unconventional for the `fetchHttpStream` to take in a function for transforming errors in the first place. Instead I would just catch the error in the place where it is invoked and then do the transformation there if any error is thrown.

Changed this. Added additional `transformError` utility function to transform the error. Alternative way would be decrease significantly the use of our async helpers (like `map` and `forEach`) and implement this in-place.

teogeb committed:
Merge branch 'main' into client-fetchHttpStream-return-lines

teogeb committed:
Merge branch 'client-fetchHttpStream-return-lines' into client-fetch-stream-abort

teogeb committed:
Merge branch 'main' into client-fetch-stream-abort

teogeb committed:
Merge branch 'main' into client-fetch-stream-abort

teogeb committed:
style
